test/parallel/test-child-process-spawnsync-validation-errors.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const spawnSync = require('child_process').spawnSync;
+
+function pass(option, value) {
+  // Run the command with the specified option. Since it's not a real command,
+  // spawnSync() should run successfully but return an ENOENT error.
+  const child = spawnSync('not_a_real_command', { [option]: value });
+
+  assert.strictEqual(child.error.code, 'ENOENT');
+}
+
+function fail(option, value, message) {
+  assert.throws(() => {
+    spawnSync('not_a_real_command', { [option]: value });
+  }, message);
+}
+
+{
+  // Validate the cwd option
+  const err = /TypeError: "cwd" must be a string/;
+
+  pass('cwd', undefined);
+  pass('cwd', null);
+  pass('cwd', __dirname);
+  fail('cwd', 0, err);
+  fail('cwd', 1, err);
+  fail('cwd', true, err);
+  fail('cwd', false, err);
+  fail('cwd', [], err);
+  fail('cwd', {}, err);
+}
+
+{
+  // Validate the detached option
+  const err = /TypeError: "detached" must be a boolean/;
+
+  pass('detached', undefined);
+  pass('detached', null);
+  pass('detached', true);
+  pass('detached', false);
+  fail('detached', 0, err);
+  fail('detached', 1, err);
+  fail('detached', __dirname, err);
+  fail('detached', [], err);
+  fail('detached', {}, err);
+}
+
+{
+  // Validate the uid option
+  if (!(common.isWindows && process.getuid() === 0)) {
+    const err = /TypeError: "uid" must be an integer/;
+
+    pass('uid', undefined);
+    pass('uid', null);
+    pass('uid', process.getuid());
+    fail('uid', __dirname, err);
+    fail('uid', true, err);
+    fail('uid', false, err);
+    fail('uid', [], err);
+    fail('uid', {}, err);
+  }
+}
+
+{
+  // Validate the gid option
+  if (!(common.isWindows && process.getgid() === 0)) {
+    const err = /TypeError: "gid" must be an integer/;
+
+    pass('gid', undefined);
+    pass('gid', null);
+    pass('gid', process.getgid());
+    fail('gid', __dirname, err);
+    fail('gid', true, err);
+    fail('gid', false, err);
+    fail('gid', [], err);
+    fail('gid', {}, err);
+  }
+}
+
+{
+  // Validate the shell option
+  const err = /TypeError: "shell" must be a boolean or string/;
+
+  pass('shell', undefined);
+  pass('shell', null);
+  pass('shell', false);
+  fail('shell', 0, err);
```

Comment:
`0` and `1` are usually acceptable as truthy values, I think they should be accepted here.